core/amber/src/main/scala/edu/uci/ics/texera/web/resource/GmailResource.scala

```diff
@@ -0,0 +1,122 @@
+package edu.uci.ics.texera.web.resource
+
+import com.google.api.client.googleapis.auth.oauth2.{
+  GoogleAuthorizationCodeTokenRequest,
+  GoogleIdTokenVerifier,
+  GoogleRefreshTokenRequest
+}
+import com.google.api.client.http.javanet.NetHttpTransport
+import com.google.api.client.json.gson.GsonFactory
+import com.google.api.services.gmail.Gmail
+import com.google.api.services.gmail.model.Message
+import com.google.auth.http.HttpCredentialsAdapter
+import com.google.auth.oauth2.{AccessToken, GoogleCredentials}
+import edu.uci.ics.amber.engine.common.AmberUtils
+import edu.uci.ics.texera.Utils
+import edu.uci.ics.texera.web.auth.SessionUser
+import io.dropwizard.auth.Auth
+import org.apache.commons.codec.binary.Base64
+
+import java.io.ByteArrayOutputStream
+import java.nio.file.Files
+import java.util.{Collections, Properties}
+import javax.annotation.security.RolesAllowed
+import javax.mail.Session
+import javax.mail.internet.{InternetAddress, MimeMessage}
+import javax.ws.rs._
+
+case class EmailMessage(email: String, subject: String, content: String)
+@Path("/gmail")
+class GmailResource {
+  final private lazy val path =
+    Utils.amberHomePath.resolve("src").resolve("main").resolve("resources").resolve("gmail")
+  final private lazy val clientId = AmberUtils.amberConfig.getString("user-sys.google.clientId")
+  final private lazy val clientSecret =
+    AmberUtils.amberConfig.getString("user-sys.google.clientSecret")
+
+  @GET
+  @RolesAllowed(Array("ADMIN"))
+  @Path("/email")
```

Comment:
The names of the APIs here are a little misleading, especially `email`. Since all the other APIs except `send` are for setting up the sender email address, I suggest making it more explicit in the API.

Reply:
Renamed.